lib/game/game_widget.dart

```diff
@@ -102,11 +113,15 @@ class _GameWidgetState extends State<GameWidget> {
   @override
   void initState() {
     super.initState();
+    // Add the initial overlays
+    widget.visibleOverlays?.forEach((key) {
+      widget.game.overlays.add(key);
+    });
     addOverlaysListener(widget.game);
   }
 
   @override
-  void didUpdateWidget(covariant GameWidget<Game> oldWidget) {
+  void didUpdateWidget(covariant GameWidget<T> oldWidget) {
```

Comment:
we may need to update this method in the case of the change in the new prop.

Reply:
Does this really need to be a `covariant`?

Reply:
@renancaraujo need your help to answer that haaha

Reply:
btw, I have done the update

Reply:
I dont think covariant is necessary here. I think autocomplete added.

Reply:
it would only make sense if you could change the type T after created, so agree it can probably be removed (afaik)